File: com.seaf.core.web/src/main/webapp/resources/js/services/rest/UserGroupService.js
(function () {

    var injectParams = ['$http', '$q', 'Base64'];

    var userFactory = function ($http, $q, Base64) {
        var serviceBaseUser = 'http://localhost:8080/com.seaf.core.service/api/v1/user/',
        	serviceBaseGroup = 'http://localhost:8080/com.seaf.core.service/api/v1/group/',
        	username = "admin",
        	password = "secret", //TODO A proteger
            factory = {};

        $http.defaults.headers.common['Authorization'] = 'Basic ' + Base64.encode(username + ':' + password);

        factory.insertUser = function (user) {
            return $http.post(serviceBaseUser, user).then(function (results) {
            	user.id = results.data.id;
                return results.data;
            });
        };

        factory.updateUser = function (user) {
            return $http.put(serviceBaseUser + user.id, user).then(function (status) {
                return status.data;
            });
        };

        factory.deleteUser = function (id) {
            return $http.delete(serviceBaseUser + id).then(function (status) {
                return status.data;
            });
        };

        factory.getUser = function (id) {
            return $http.get(serviceBaseUser + id).then(function (results) {
                return results.data;
            });
        };

        factory.getUsers = function (page, size, search, sort, dir) {
        	if(search === undefined||search === null)
        		search = '';
        	
            return $http.get(serviceBaseUser 
            		+ '?page=' + page 
            		+ '&size=' + size 
            		+ '&search=' + search
            		+ '&sort=' + sort
            		+ '&dir=' + dir).then(function (response) {
            	return response.data;
            });
        };
        
        factory.insertGroup = function (group) {
            return $http.post(serviceBaseGroup, group).then(function (results) {
            	group.id = results.data.id;
                return results.data;
            });
        };

        factory.updateGroup = function (group) {
            return $http.put(serviceBaseGroup + group.id, group).then(function (status) {
                return status.data;
            });
        };

        factory.deleteGroup = function (id) {
            return $http.delete(serviceBaseGroup + id).then(function (status) {
                return status.data;
            });
        };

        factory.getGroup = function (id) {
            return $http.get(serviceBaseGroup + id).then(function (results) {
                return results.data;
            });
        };

        factory.getGroups = function (page, size, search, sort, dir) {
        	if(search === undefined||search === null)
        		search = '';
        	
            return $http.get(serviceBaseGroup 
            		+ '?page=' + page 
            		+ '&size=' + size 
            		+ '&search=' + search
            		+ '&sort=' + sort
            		+ '&dir=' + dir).then(function (response) {
            	return response.data;
            });
        };
        
        factory.subscribeGroup = function (userid, groupid) {
            return $http.put(serviceBaseUser + userid + "/group/" + groupid).then(function (status) {
                return status.data;
            });
        };
        
        factory.unsubscribeGroup = function (userid, groupid) {
            return $http.delete(serviceBaseUser + userid + "/group/" + groupid).then(function (status) {
                return status.data;
            });
        };        

        return factory;
    };

    userFactory.$inject = injectParams;

    angular.module('SeafApp').factory('userService', userFactory);

}());

File: com.seaf.core.service/src/main/java/com/seaf/core/service/business/UserGroupService.java
package com.seaf.core.service.business;

import java.util.List;

import com.seaf.core.service.business.exception.UserGroupException;
import com.seaf.core.service.model.GroupDto;
import com.seaf.core.service.model.UserDto;
import com.seaf.core.service.model.utils.EnvelopeList;

public interface UserGroupService {
	
	public EnvelopeList 	getUsers		(String searchQuery, int pageNumber, int pageSize, String sortAttribute, String sortDirection);
	public UserDto 			getUser			(int userId) 					throws UserGroupException;
	public UserDto 			addUser			(UserDto user)  				throws UserGroupException;
	public UserDto 			modifyUser		(int userId, UserDto user)		throws UserGroupException;
	public void 			deleteUser		(int userId)					throws UserGroupException;

	public EnvelopeList 	getGroups		(String searchQuery, int pageNumber, int pageSize, String sortAttribute, String sortDirection);
	public GroupDto 		getGroup		(int groupId) 					throws UserGroupException;
	public GroupDto 		addGroup		(GroupDto group)  				throws UserGroupException;
	public GroupDto 		modifyGroup		(int groupId, GroupDto group)	throws UserGroupException;
	public void 			deleteGroup		(int groupId)					throws UserGroupException;
	
	public GroupDto			subscribeGroup	(int groupId, int userId)		throws UserGroupException;
	public void				unsubscribeGroup(int groupId, int userId)		throws UserGroupException;
	
	public List<GroupDto>	getGroupOfUser	(int userId)					throws UserGroupException;
	public List<UserDto> 	getUserOfGroup	(int groupId) 					throws UserGroupException;
}
